### ISSUE TITLE
Improve perfornace of GraphQL call for widget

### ISSUE THREAD

**Issue by kjgarza** [datacite/spitz#6]
Currently graphql for data metrics takes around 4s. We need to improve it. <br/>
 Optimal response time: TBD <br/> API Limit: TBD  <br/>  
Tasks: <br/> 
- [x] Avoid script usage <br/>
 - [x] Evaluate pre-indexing for citations. Will need to reindex the whole eventData index.  <br/>

 child https://github.com/datacite/datacite/issues/802

**Comment by kjgarza** [comment]
https://www.elastic.co/guide/en/elasticsearch/reference/master/tune-for-search-speed.html#_pre_index_data

> Search as few fields as possibleedit
> The more fields a query_string or multi_match query targets, the slower it is. A common technique to improve search speed over multiple fields is to copy their values into a single field at index time, and then use this field at search time. This can be automated with the copy-to directive of mappings without having to change the source of documents. Here is an example of an index containing movies that optimizes queries that search over both the name and the plot of the movie by indexing both values into the name_and_plot field
>
> Pre-index dataedit
> You should leverage patterns in your queries to optimize the way data is indexed. For instance, if all your documents have a price field and most queries run range aggregations on a fixed list of ranges, you could make this aggregation faster by pre-indexing the ranges into the index and using a terms aggregations.

According to our criteria, Every Event can be at most two `types of links` at the same time:

- An event of the type `citation-link` will always be also  `reference-link` at the same time. A `citation-link` for the object of the event, it's a  `reference-link` for the object and vice-versa.
- An event of the type  `relation-link` can only be a  `reference-link` type only.

**Comment by kjgarza** [comment]
as mentioned in the PR in https://github.com/datacite/lupo/pull/369 a discussion of the whole architecture would have to happen before working in improving the performance. 

As for MVP the we will work with the current performance

**Comment by kjgarza** [comment]
closing in favor of https://github.com/datacite/datacite/issues/921 where all the data pre-indexing activities will take place.

Once that is done, metrics counts will come from there rather than from queries.
